pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -1420,9 +1435,16 @@ func (ctrl *PersistentVolumeController) provisionClaimOperation(claim *v1.Persis
 	}
 
 	opComplete := util.OperationCompleteHook(plugin.GetPluginName(), "volume_provision")
+	// TODO: modify the Provision() interface to pass in the allowed topology information
+	// of the provisioned volume.
 	volume, err = provisioner.Provision()
 	opComplete(&err)
 	if err != nil {
+		// Other places of failure has nothing to do with DynamicProvisioningScheduling,
+		// so just let controller retry in the next sync. We'll only call func
+		// triggerReprovisioning here when the underlying provisioning actually failed.
+		ctrl.triggerReprovisioning(claim)
```

Comment:
A better name here might be "rescheduleProvisioning"

Reply:
It would also be good (in a separate PR), to have an integration test for this interaction between scheduler, pv controller, and provisioner.

Reply:
Yeah, I meant to add the test after the base PR merged